src/esp/io/URDFParser.h

```diff
@@ -0,0 +1,596 @@
+    if (m_linkIndicesToNames.count(linkIndex) != 0u) {
+      return getLink(m_linkIndicesToNames.at(linkIndex))->m_parentJoint.lock();
+    }
+    return nullptr;
+  }
+
+  Model() = default;
+
+  /**
+   * @brief Set global scaling and re-scale an existing model. Modifies various
+   * internal parameters.
+   *
+   * @param scaling The new absolute uniform scale.
+   */
+  void setGlobalScaling(float scaling) {
+    if (scaling == m_globalScaling) {
+      // do nothing
+      return;
+    }
+
+    // Need to re-scale model, so use the ratio of new to current scale
+    float scaleCorrection = scaling / m_globalScaling;
+
+    // scale all transforms' translations
+    for (const auto& link : m_links) {
+      // scale inertial offsets
+      link.second->m_inertia.m_linkLocalFrame.translation() *= scaleCorrection;
+      // scale visual shape parameters
+      for (auto& visual : link.second->m_visualArray) {
+        scaleShape(visual, scaleCorrection);
+      }
+      // scale collision shapes
+      for (auto& collision : link.second->m_collisionArray) {
+        scaleShape(collision, scaleCorrection);
+      }
+    }
+    for (const auto& joint : m_joints) {
+      // scale joint offsets
+      joint.second->m_parentLinkToJointTransform.translation() *=
+          scaleCorrection;
+      // scale prismatic joint limits
+      if (joint.second->m_type == PrismaticJoint) {
+        joint.second->m_lowerLimit *= double(scaleCorrection);
+        joint.second->m_upperLimit *= double(scaleCorrection);
+      }
+    }
+
+    m_globalScaling = scaling;
+  }
+
+  //! Get the currently configured global model scaling
+  float getGlobalScaling() const { return m_globalScaling; }
+
+  /**
+   * @brief Set scaling for mass from initial values configured in URDF.
+   * Modifies various internal parameters.
+   *
+   * @param massScaling The new absolute uniform mass scale.
+   */
+  void setMassScaling(float massScaling) {
+    if (massScaling == m_massScaling) {
+      // do nothing
+      return;
+    }
+    float massScaleCorrection = massScaling / m_massScaling;
+
+    // Only need to scale the per-link mass values. These will be further
+    // processed during import.
+    for (const auto& link : m_links) {
+      Inertia& linkInertia = link.second->m_inertia;
+      linkInertia.m_mass *= double(massScaleCorrection);
+    }
+
+    m_massScaling = massScaling;
+  }
+
+  //! Get the currently configured mass scaling of the model
+  float getMassScaling() const { return m_massScaling; }
+
+ protected:
+  // scaling values which can be applied to the model after parsing
+  //! Global euclidean scaling applied to the model's transforms, asset scales,
+  //! and prismatic joint limits. Does not affect mass.
+  float m_globalScaling = 1.0;
+
+  //! Mass scaling of the model's Link inertias.
+  float m_massScaling = 1.0;
+
+  //! Scale the transformation and parameters of a Shape
+  void scaleShape(Shape& shape, float scale) {
```

Comment:
Should this be in the cpp, not the header?

Counterpoint: I can appreciate not changing this class too much, in case we ever want to diff against the version inside Bullet.

Reply:
This one is my function encapsulating all the changes necessary to modify the scale after parsing. Sure we can move it into the cpp.